blog/2018-08-30-streaming-mysql-data-changes-into-kinesis.adoc

```diff
@@ -0,0 +1,411 @@
+In the following we're going to use the connector for MySQL.
+We also need to add a dependency to the https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/kinesis/package-summary.html[Kinesis Client API], so these are the dependencies needed:
+
+[source,xml]
+----
+...
+<dependency>
+    <groupId>io.debezium</groupId>
+    <artifactId>debezium-embedded</artifactId>
+    <version>0.8.1.Final</version>
+</dependency>
+<dependency>
+    <groupId>io.debezium</groupId>
+    <artifactId>debezium-connector-mysql</artifactId>
+    <version>0.8.1.Final</version>
+</dependency>
+<dependency>
+    <groupId>com.amazonaws</groupId>
+    <artifactId>amazon-kinesis-client</artifactId>
+    <version>1.9.0</version>
+</dependency>
+...
+----
+
+== Configuring the Embedded Engine
+
+The Debezium embedded engine is configured through an instance of `io.debezium.config.Configuration`.
+This class can obtain values from system properties or from a given config file,
+but for the sake of the example we'll simply pass all required values via its fluent builder API:
+
+[source,java]
+----
+Configuration config = Configuration.create()
+    .with(EmbeddedEngine.CONNECTOR_CLASS, "io.debezium.connector.mysql.MySqlConnector")
+    .with(EmbeddedEngine.ENGINE_NAME, "kinesis")
+    .with(MySqlConnectorConfig.SERVER_NAME, "kinesis")
+    .with(MySqlConnectorConfig.SERVER_ID, 8192)
+    .with(MySqlConnectorConfig.HOSTNAME, "localhost")
+    .with(MySqlConnectorConfig.PORT, 3306)
+    .with(MySqlConnectorConfig.USER, "debezium")
+    .with(MySqlConnectorConfig.PASSWORD, "dbz")
+    .with(MySqlConnectorConfig.DATABASE_WHITELIST, "inventory")
+    .with(MySqlConnectorConfig.TABLE_WHITELIST, "inventory.customers")
+    .with(EmbeddedEngine.OFFSET_STORAGE, "org.apache.kafka.connect.storage.MemoryOffsetBackingStore")
+    .with(MySqlConnectorConfig.DATABASE_HISTORY, MemoryDatabaseHistory.class.getName())
+    .with("schemas.enable", false)
+    .build();
+----
+
+If you've ever set up the Debezium MySQL connector in Kafka Connect, most of the properties will look familiar to you.
+
+But let's talk about the `OFFSET_STORAGE` and `DATABASE_HISTORY` options in a bit more detail.
+They deal with how connector offsets and the database history should be persisted.
+When running the connector via Kafka Connect, both would typically be stored in specific Kafka topics.
+But that's not an option here, so an alternative is needed.
+For this example we're simply going to keep the offsets and database history in memory.
+I.e. if the engine is restarted, this information will be lost and the connector will start from scratch, e.g. with a new initial snapshot.
+
+While out of scope for this blog post, it wouldn't be too difficult to create alternative implementations of the `OffsetBackingStore` and `DatabaseHistory` contracts, respectively.
```

Comment:
I think file based stores can get honorable mention as they are available for limit ed produciton use cases

Reply:
Ok, will do.